msi/createmsi.py

```diff
@@ -78,78 +35,180 @@ def __init__(self, dirs, files):
-            self.platform_sr = ''
             self.progfile_dir = 'ProgramFilesFolder'
-            self.component_platform = ''
+        self.component_num = 0
+        self.feature_properties = {
+            self.staging_dirs[0]: {
+                'Id': 'MainProgram',
+                'Title': 'Meson',
+                'Description': 'Meson executables',
+                'Level': '1',
+                'Absent': 'disallow', 
+            },
+            self.staging_dirs[1]: {
+                'Id': 'NinjaProgram',
+                'Title': 'Ninja',
+                'Description': 'Ninja build tool',
+                'Level': '1',
+            }
+        }
+        self.feature_components = {}
+        for sd in self.staging_dirs:
+            self.feature_components[sd] = []
 
     def build_dist(self):
-        if os.path.exists(self.staging_dir):
-            shutil.rmtree(self.staging_dir)
+        for sdir in self.staging_dirs:
+            if os.path.exists(sdir):
+                shutil.rmtree(sdir)
+        main_stage, ninja_stage = self.staging_dirs
         modules = [os.path.splitext(os.path.split(x)[1])[0] for x in glob(os.path.join('mesonbuild/modules/*'))]
         modules = ['mesonbuild.modules.' + x for x in modules if not x.startswith('_')]
         modulestr = ','.join(modules)
         subprocess.check_call(['c:\\Python\python.exe',
                                'c:\\Python\Scripts\\cxfreeze',
                                '--target-dir',
-                               self.staging_dir,
+                               main_stage,
                                '--include-modules',
                                modulestr,
                                'meson.py'])
-        shutil.copy(shutil.which('ninja'), self.staging_dir)
-        if not os.path.exists(os.path.join(self.staging_dir, 'meson.exe')):
+        if not os.path.exists(os.path.join(main_stage, 'meson.exe')):
             sys.exit('Meson exe missing from staging dir.')
-        if not os.path.exists(os.path.join(self.staging_dir, 'ninja.exe')):
+        os.mkdir(ninja_stage)
+        shutil.copy(shutil.which('ninja'), ninja_stage)
+        if not os.path.exists(os.path.join(ninja_stage, 'ninja.exe')):
             sys.exit('Ninja exe missing from staging dir.')
 
-
     def generate_files(self):
-        assert(os.path.isdir(self.staging_dir))
-        comp_ref_xml = ''
-        nodes = {}
-        with open(self.main_xml, 'w') as ofile:
-            for root, dirs, files in os.walk(self.staging_dir):
+        self.root = ET.Element('Wix', {'xmlns': 'http://schemas.microsoft.com/wix/2006/wi'})
+        product = ET.SubElement(self.root, 'Product', {
+            'Name': self.product_name,
+            'Manufacturer': 'The Meson Development Team',
+            'Id': self.guid,
+            'UpgradeCode': self.update_guid,
+            'Language': '1033',
+            'Codepage':  '1252',
+            'Version': self.version,
+            })
```

Comment:
__[flake8]__
* [E123] closing bracket does not match indentation of opening bracket's line

<!-- Comment by [SideCI](https://sideci.com) -->